src/WebJobs.Script/Rpc/Capabilities.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Google.Protobuf.Collections;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    internal class Capabilities
+    {
+        private readonly ILogger _logger;
+        private IDictionary<string, string> _capabilities = new Dictionary<string, string>();
+
+        public Capabilities(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public string GetCapabilityState(string capability)
+        {
+            if (_capabilities.TryGetValue(capability, out string state))
+            {
+                return state;
+            }
+            return null;
+        }
+
+        public void UpdateCapabilities(MapField<string, string> capabilities)
+        {
+            if (capabilities == null)
+            {
+                return;
+            }
+
+            _logger.LogInformation($"Requested capabilities: {capabilities.ToString()}");
```

Comment:
update to  `_logger.LogDebug("Updating capabilities: {capabilities}",capabilities.ToString());`

Reply:
capabilities.ToString() does this print key and value? If now, better to log in UpdateCapability method.

Reply:
It does log key-value pairs.